pxtcompiler/emitter/service.ts

```diff
@@ -1088,6 +1093,13 @@ namespace ts.pxtc.service {
                         tsPos = smallest.ts.startPos
                     }
                 }
+
+                // filter based on word match if we get too many (ideally we'd leave this filtering for monaco as it's
+                // better at fuzzy matching and fluidly changing but for performance reasons we want to do it here)
+                if (!isMemberCompletion && resultSymbols.length > MAX_SYMBOLS_BEFORE_FILTER) {
```

Comment:
I think this will help. It looks like the issue is primarily in Monaco. Profiling it, all the time is going to rendering the suggestion elements.